src/ua.c

```diff
@@ -954,18 +1142,33 @@ int ua_connect_dir(struct ua *ua, struct call **callp,
 	struct pl pl;
 	int err = 0;
 
-	if (!ua || !str_isset(req_uri))
+	if (!str_isset(req_uri))
 		return EINVAL;
 
 	dialbuf = mbuf_alloc(64);
 	if (!dialbuf)
 		return ENOMEM;
 
+	if (!ua) {
```

Comment:
this logic should be done outside ua.c

Reply:
Ok, I'll move this to menu. But then gtk has to do it similar.